src/utils.js

```diff
@@ -392,7 +392,15 @@ const maskOptions = {
 };
 
 export const formatMask = (value, mask) => {
-  const text = (value === null || value === undefined) ? '' : `${value}`;
+  let text = (value === null || value === undefined) ? '' : `${value}`;
+
+  if (mask === '(999) 999-9999') {
+    if (text.indexOf('-') || text.indexOf(' ')) {
+      // remove hyphen and space characters
+      text = text.replace('-', '').replace(' ', '');
+    }
+  }
+
```

Comment:
I like the idea of removing the characters before applying the mask but I think that shouldn't be placed here. Instead, lets add a third argument for maskFilter. The default filter can be:

```
const defaultParser = value => ((value === null || value === undefined) ? '' : `${value}`);

export const formatMask = (value, mask, maskParser) => {
  const parse = maskParser || defaultParser;
  let text = parse(value);
  // ...
}
```

We also need to change `TextInputWidget` to accept `maskParser` and change  `PhoneWidget` to send `maskParser`.

the parser can be a bit more general:
```
const maskParser = (value) => {
  const text = (value === null || value === undefined) ? '' : `${value}`;
  return text.replace(/[^0-9]/g, '');
};
```